Sources/Purchasing/Offerings.swift

```diff
@@ -93,4 +96,48 @@ public extension Offerings {
         return description
     }
 
+    /**
+     Retrieves a current offering for a placement identifier, use this to access offerings defined by targeting
+     placements configured in the RevenueCat dashboard, 
+     e.g. `offerings.getCurrentOffering(forPlacement: "placement_id")`.
+     */
+    @objc(getCurrentOfferingForPlacement:)
+    func getCurrentOffering(forPlacement placementIdentifier: String) -> Offering? {
+        guard let placements = self.placements else {
+            return nil
+        }
+
+        let returnOffering: Offering?
+        if let explicitOfferingId: String? = placements.offeringIdsByPlacement[placementIdentifier] {
+            // Don't use fallback since placement id was explicity set in the dictionary
+            returnOffering = explicitOfferingId.flatMap { self.all[$0] }
+        } else {
+            // Use fallback since the placement didn't exist
+            returnOffering =  placements.fallbackOfferingId.flatMap { self.all[$0]}
+        }
+
+        return returnOffering?.copyWith(placementIdentifier: placementIdentifier)
+    }
+}
+
+private extension Offering {
+    func copyWith(placementIdentifier: String?) -> Offering {
+        let updatedPackages = self.availablePackages.map { pkg in
+            let newContext = PresentedOfferingContext(
+                offeringIdentifier: pkg.presentedOfferingContext.offeringIdentifier,
+                placementIdentifier: placementIdentifier
+            )
+            return Package(identifier: pkg.identifier,
+                           packageType: pkg.packageType,
+                           storeProduct: pkg.storeProduct,
+                           presentedOfferingContext: newContext
```

Comment:
Nitpick but should we have an internal copy constructor from package so we can create a new package with a new context without having to pass all the parameters again?